### PR TITLE
add hasNotificationsPermissions and internalNavigation methods

### PR DESCRIPTION
**Pull request by andyindahouse**
Look at "OS Permission Status" and "Internal Navigation" points.

https://docs.google.com/document/d/1kzF2d1LXftPdGpxYst_UOFiQunRI1T6CiNmCLDC6_MM/edit#

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -496,6 +496,49 @@ onNativeEvent(({event}) => {
...
+hasNotificationsPermissions().then((hasNotifications) => {
+    console.log(hasNotifications);
+}).catch(err => {
+    console.error(err);
+};
+```
+
+### internalNavigation
+
+Init a internal and native navigation to a feature specific of a device
```
Comment (pladaria) [inline README.md]: maybe better:
```
Init an internal and native navigation to a device specific feature
```

File: README.md [hunk 2]
```diff
@@ -496,6 +496,49 @@ onNativeEvent(({event}) => {
...
+
+-   Available for app versions 11.4 and higher
+
+```typescript
+internalNavigation: (feature: string) => Promise<void>;
```
Comment (pladaria) [inline README.md]: document available features

File: src/device.ts [hunk 3]
```diff
@@ -16,3 +16,12 @@ export const requestDeviceImei = (): Promise<string | null> =>
     postMessageToNativeApp({type: 'IMEI'}, TIMEOUT)
         .then(({imei}) => imei)
         .catch(() => null);
+
+export const internalNavigation = (feature: string): Promise<void> => {
+    return postMessageToNativeApp({
```
Comment (pladaria) [inline src/device.ts]: unneeded return, you can just:
```js
export const internalNavigation = (...): Promise<void> => postMessageToNativeApp({...});
```

File: src/webview-bridge-cjs.js.flow [hunk 4]
```diff
@@ -65,6 +65,8 @@ declare export function requestSimImsi(): Promise<string | null>;
 
 declare export function requestDeviceImei(): Promise<string | null>;
 
+declare export function internalNavigation(feature: string): Promise<void>;
```
Comment (pladaria) [inline src/webview-bridge-cjs.js.flow]: type param with available features:
```js
internalNavigation(feature: 'internal-navigation')
```

File: src/permission-status.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import {postMessageToNativeApp} from './post-message';
+
+const TIMEOUT = 200;
+
+export const hasNotificationsPermissions = (): Promise<boolean> =>
```
Comment (pladaria) [inline src/permission-status.ts]: I'd keep this function generic (receive `feature` and `params` by parameters), name it for example: `checkPermissionStatus(feature: string, params: {[string]: string})`

In webapp side you can write the semantic method (`hasNotificationPermission`)

Also, don't create a new file for this, you can put this function inside utils

File: src/webview-bridge-cjs.js.flow (hunk 4 above)
Comment (andyindahouse) [reply src/webview-bridge-cjs.js.flow]: I've talked with Yamal and I see that right now it only support 'notification-settings' as feature, so I'll change this

File: README.md [hunk 6]
```diff
@@ -496,6 +496,58 @@ onNativeEvent(({event}) => {
...
+};
+```
+
+### internalNavigation
+
+Init an internal and native navigation to a device specific feature
+
+-   Available for app versions 11.4 and higher
+
+You have these features avalaibles:
```
Comment (pladaria) [inline README.md]: `You have these features available:` (singular)

or just:

`Available features:`

Same in the other method

File: src/utils.ts [hunk 7]
```diff
@@ -152,3 +152,24 @@ export const fetch = ({
...
+            type: 'OS_PERMISSION_STATUS',
+            payload: {
+                feature,
+                params: params,
+            },
+        },
+        TIMEOUT,
+    )
+        .then(({granted}) => granted)
+        .catch(() => false);
```
Comment (pladaria) [inline src/utils.ts]: don't catch here, lib users may want to know the reason of failure, do that in webapp

File: README.md [hunk 8]
```diff
@@ -496,6 +496,58 @@ onNativeEvent(({event}) => {
...
+
+#### Example
+
+```javascript
+import {internalNavigation} from '@tef-novum/webview-bridge';
+
+internalNavigation('notification-settings')
+.catch(err => {
+    console.error(err);
+};
```
Comment (pladaria) [inline README.md]: I'd remove this from the example

File: README.md [hunk 9]
```diff
@@ -496,6 +496,58 @@ onNativeEvent(({event}) => {
...
+#### Example
+
+```javascript
+import {checkPermissionStatus} from '@tef-novum/webview-bridge';
+
+checkPermissionStatus('notifications',{channelId: 'default'}).then((hasPermissions) => {
+    console.log(hasPermissions);
+}).catch(err => {
+    console.error(err);
+};
```
Comment (pladaria) [inline README.md]: I'd remove the catch from the example

File: src/utils.ts [hunk 10]
```diff
@@ -152,3 +152,24 @@ export const fetch = ({
...
+): Promise<boolean> =>
+    postMessageToNativeApp(
+        {
+            type: 'OS_PERMISSION_STATUS',
+            payload: {
+                feature,
+                params: params,
+            },
+        },
+        TIMEOUT,
```
Comment (pladaria) [inline src/utils.ts]: is this timeout needed for any reason?

Comment (andyindahouse) [reply src/utils.ts]: hmm, I think that not, I'll remove it

**Pull request merged**